common/src/main/java/com/wynntils/wynn/utils/WynnBossBarUtils.java

```diff
@@ -7,38 +7,45 @@
 import com.wynntils.mc.utils.ComponentUtils;
 import com.wynntils.mc.utils.McUtils;
 import com.wynntils.utils.Pair;
+import java.util.Collection;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import net.minecraft.client.gui.components.LerpingBossEvent;
 
 public class WynnBossBarUtils {
-    public static final ManaBank NO_MANA_BANK = new ManaBank(0, 0, -1);
-    public static final BloodPool NO_BLOOD_POOL = new BloodPool(0, -1);
-    public static final AwakenedBar NO_AWAKENED_PROGRESS = new AwakenedBar(0, 0, -1);
-    public static final Focus NO_FOCUS = new Focus(0, 0, -1);
+    public static final BarProgress NO_MANA_BANK = new BarProgress(0, 0, -1);
+    public static final BarProgress NO_BLOOD_POOL = new BarProgress(0, 0, -1);
+    public static final BarProgress NO_AWAKENED_PROGRESS = new BarProgress(0, 0, -1);
+    public static final BarProgress NO_FOCUS = new BarProgress(0, 0, -1);
 
     private static final Pattern BLOOD_POOL_PATTERN = Pattern.compile("§cBlood Pool §4\\[§c(\\d+)%§4\\]");
     private static final Pattern MANA_BANK_PATTERN = Pattern.compile("§bMana Bank §3\\[(\\d+)/(\\d+)§3\\]");
     private static final Pattern AWAKENED_PROGRESS_PATTERN = Pattern.compile("§fAwakening §7\\[§f(\\d+)/(\\d+)§7]");
     private static final Pattern FOCUS_PATTERN = Pattern.compile("§eFocus §6\\[§e(\\d+)/(\\d+)§6]");
 
-    private static Pair<LerpingBossEvent, Matcher> getLerpingBossEvent(Pattern titlePattern) {
+    private static Pair<LerpingBossEvent, Matcher> getLerpingBossEvent(Pattern titlePattern, boolean cancel) {
         LerpingBossEvent poolEvent = null;
         Matcher matcher = null;
 
-        for (LerpingBossEvent event : McUtils.mc().gui.getBossOverlay().events.values()) {
+        Collection<LerpingBossEvent> events =
+                McUtils.mc().gui.getBossOverlay().events.values();
+
+        for (LerpingBossEvent event : events) {
             matcher = titlePattern.matcher(ComponentUtils.getCoded(event.getName()));
             if (matcher.matches()) {
                 poolEvent = event;
+                if (cancel) {
+                    events.remove(event);
```

Comment:
Well, this is kinda weird. But this sure does its job and is not that intrusive as my solution with redirect.

Reply:
I'm not really following where and how this is happening in time. So the boss bar gets data gets updated in vanilla, but then this method is called, and it removes the boss bar from the set of boss bars? But whenever Vanilla gets a new boss bar update packet, it will recreate the data? What if Vanilla happens to render the bars before this method is called? Will the boss bar flicker in and out?

I agree that this is less intrusive, but I must feel convinced it is stable and reliable.

Reply:
I have yet to fully test this yet. I hope it does not flicker serendipidiously :-), but if it does, I may have to resort to more intrusive methods

Reply:
EDIT: This does not work sadly